docs/learning/backend/Workspace.md

```diff
@@ -444,32 +257,34 @@ Possible resource types are:
 - `blob` resources that hold `Uint8Array`s. They may be loaded with [WorkspaceDb.getBlob]($backend).
 - `file` resources that hold arbitrary files. They may be extracted to local files with [WorkspaceDb.getFile]($backend)
 
-> Note: files may be compressed as they are stored in `WorkspaceContainer`s.
+> Note: files may be compressed as they are stored in `WorkspaceContainer`s. In general, files should only be used for backwards compatibility with code that reads files from disk. They should be avoided if possible.
 
 ### WorkspaceResource.Names
 
 [WorkspaceResource.Name]($backend)s identify resources within a `WorkspaceDb`. There are no restrictions on the format of a `WorkspaceResource.Name`, other than they are limited to 1024 characters and may not start or end with a blank character.
 
 > Note: `WorkspaceResource.Name`s must be unique for each resource type. But, it is possible to have a `string`, a `blob`, and a `file` resource in the same `WorkspaceDb` with the same `WorkspaceResource.Name`.
```

Comment:
> it is possible to have a `string`, a `blob`, and a `file` resource in the same `WorkspaceDb` with the same `WorkspaceResource.Name`

Do you have use cases for this? It seems confusing and error-prone.

Reply:
No, I don't really have a use case. But they are stored in 3 different tables, so there's no way to enforce uniqueness across them without a schema change adding some other "name" table". I don't really think its a big deal, people have to know what type their resource is.